src/osiris_replica_reader.erl

```diff
@@ -84,20 +97,26 @@ init(#{host := Host,
        connection_token := Token} =
          Args) ->
     process_flag(trap_exit, true),
-    CntId = {?MODULE, ExtRef, Host, Port},
-    CntSpec = {CntId, ?COUNTER_FIELDS},
-    %% TODO: handle errors
-    {ok, Log} = osiris_writer:init_data_reader(LeaderPid, TailInfo, CntSpec),
-    CntRef = osiris_log:counters_ref(Log),
-    ?INFO("starting replica reader ~s at offset ~b Args: ~p",
-          [Name, osiris_log:next_offset(Log), Args]),
+
+
+
     SndBuf = 146988 * 10,
-    {ok, Sock} =
-        gen_tcp:connect(Host, Port,
+
+    {ok, Sock, Host} =
```

Comment:
can we handle this with a case and return {stop, Reason} from the init function if we don't manage to connect? nicer than just crashing.